aws-cloudformation/aws-cloudformation-resource-providers-ecr
# add RepositoryUri readonly property for use in GetAtt

*Issue #, if available:*
https://github.com/aws-cloudformation/aws-cloudformation-coverage-roadmap/issues/124

*Description of changes:*
Currently, the repository resource only supports the Arn for use with Fn::GetAtt.
This change adds the repositoryUri as a readonly property so that it can be used as well.

*Testing*
- mvn package
- cfn test
```
handler_create.py::contract_create_delete PASSED [  6%]
handler_create.py::contract_invalid_create PASSED [ 12%]
handler_create.py::contract_create_duplicate PASSED  [ 18%]
handler_create.py::contract_create_read_success PASSED [ 25%]
handler_create.py::contract_create_list_success PASSED [ 31%]
handler_delete.py::contract_delete_read PASSED  [ 37%]
handler_delete.py::contract_delete_list PASSED  [ 43%]
handler_delete.py::contract_delete_update PASSED [ 50%]
handler_delete.py::contract_delete_delete PASSED  [ 56%]
handler_delete.py::contract_delete_create PASSED  [ 62%]
handler_misc.py::contract_check_asserts_work PASSED  [ 68%]
handler_read.py::contract_read_without_create PASSED  [ 75%]
handler_update.py::contract_update_read_success PASSED [ 81%]
handler_update.py::contract_update_list_success PASSED  [ 87%]
handler_update_invalid.py::contract_update_create_only_property PASSED  [ 93%]
handler_update_invalid.py::contract_update_non_existent_resource PASSED [100%]
```
- cfn submit


- manual test with the following template:
```
Parameters:    
  repositoryName:
    Type: String
Resources:
  MyRepository:
    Type: AWS::ECR::Repository
    Properties: 
      RepositoryName: !Ref repositoryName
Outputs:    
  RepoArn:
    Value: !GetAtt MyRepository.Arn
  RepoUri:
    Value: !GetAtt MyRepository.RepositoryUri
```

- Verified creation & stack output values through `aws cloudformation describe-stacks`

- manually tested updating stack with new repo name & verified corresponding output
- manually deleted stack

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
